ystia/yorc
# Feature/gh 104 app monitoring

# Pull Request description
Monitor deployed services liveness#104
As: an app manager
I want to: know which of my deployed services are up and down
So that: I can trigger some operations to get them up

AC1: statuses could be retrieved using the REST API for an integration with other systems (may be A4C)
AC2: limited to HTTP/TCP checks for this first version
AC3: Based on TOSCA policies applied to nodes

## Description of the change
#### Yorc side
- Add **yorc.policies.Monitoring** policy type
- Add **yorc.policies.monitoring.HTTPMonitoring** policy type
- Add **yorc.policies.monitoring.TCPMonitoring** policy type
- Pass yorc-types to 1.1.0 version
- Store policies (types and templates) in Consul
- Add policies API to handle deployment policies (GetPoliciesForTypeAndNode, GetPolicyPropertyValue, GetPolicyType, IsTargetForPolicy, GetPolicyTargets, GetPolicyTargetsForType)
- Refactor monitoring hooks to lookup deployment policies and register TCP/HTTP checks (remove the previous compute TCP monitoring version)
- Add tcpCheckExecution and httpCheckExecution to execute check
#### Alien side
- Add modifier to check only one policy is applied on a node name and targets field is set and are valids
- Add documentation on how-to use HTTP monitoring policy
### How to verify it
- Follow the Alien plugin documentation to monitor Welcome Application.
- Use secured yorc topology to check HTTP monitoring with TLS client config
(for example, if you check the /health url, you need to add a HTTP header in the policy config in order to accept appllcation/json content)

### Description for the changelog
* Monitor deployed services liveness ([GH-104](https://github.com/ystia/yorc/issues/104))
## Applicable Issues
#104
